#5663 Paella7 patches frameList undefined exception

karendolan opened the pull request:
### Your pull request should…

* [1/2] have a concise title
* [x ] [close an accompanying issue](https://help.github.com/en/articles/closing-issues-using-keywords) if one exists
* [ x] [be against the correct branch](https://docs.opencast.org/develop/developer/development-process#acceptance-criteria-for-patches-in-different-versions)
* [x ] pass automated tests
* [ x] have a clean commit history
* [x ] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)

This patch fixes issues:
1. For sites that publish either "presenter" OR "presentation" or both, to have preview slides, by changing the fixed ONE type of flavor for navigation slides to be a fallback array, and also allowing the config list to be an array instead of a single string.
2. The paella-core issue of https://github.com/polimediaupv/paella-core/issues/343 that throws a fatal undefined exception when the manifest.frameList is undefined, by sending an empty array when neither the config NOR the fallback presenter NOR presentation has any "segment+preview" flavored images
3. Corrects Karen Dolan's committer email in the main pom
4. Adds a question about the recursion loop `f4454f61f2f (Dennis Benz         2023-07-20 21:29:55 +0200 295)       videoPreview = getVideoPreview(episode.mediapackage, config);`

karendolan committed:
Issue #5663 fallback navigation slide flavors as an array and Paella-core issue #343 manifest.frameList undefined exception
Fix contrib email address for karen dolan
Peer review updates on pull
Peer review change to target 15.x
added the nav frame video target content so nav frames can find the target video to overlay

karendolan commented on modules/engage-paella-player-7/src/js/EpisodeConversor.js:
```diff
@@ -278,13 +290,15 @@ function processAttachments(episode, manifest, config) {
+      // TODO: will this cause an endless recursion loop when there are no
+      // navigation segments preview images within the mediapackage?
       videoPreview = getVideoPreview(episode.mediapackage, config);
     }
```
Ref change at https://github.com/opencast/opencast/commit/f4454f61f2f

dennis531 replied:
I don't see a problem there, because the function `getVideoPreview` doesn't call any other function recursively.

karendolan replied:
I removed that comment and clarified the names of the variables to avoid confusion about which variable is being set

dennis531 commented on modules/engage-paella-player-7/src/js/EpisodeConversor.js:
```diff
@@ -278,13 +290,15 @@ function processAttachments(episode, manifest, config) {
     else {
+      // TODO: will this cause an endless recursion loop when there are no
+      // navigation segments preview images within the mediapackage?
       videoPreview = getVideoPreview(episode.mediapackage, config);
```
The performance could be improved:
```suggestion
      if (videoPreview === null) {
        videoPreview = getVideoPreview(episode.mediapackage, config);
      }
```

karendolan replied:
I added the conditional around that call. Thanks

gregorydlogan commented:
This at least partially duplicates #5488, and should probably go into `r/15.x` anyway.

karendolan commented:
> This at least partially duplicates #5488, and should probably go into `r/15.x` anyway.

Moved patch to r/15.x and also added a related patch for finding the target video to add the navigation slide onto.
